app/src/main/java/woowacourse/shopping/cart/contract/presenter/CartPresenter.kt

```diff
@@ -0,0 +1,64 @@
+package woowacourse.shopping.cart.contract.presenter
+
+import com.domain.model.CartRepository
+import woowacourse.shopping.cart.CartItem
+import woowacourse.shopping.cart.contract.CartContract
+import woowacourse.shopping.mapper.toUIModel
+import woowacourse.shopping.model.CartUIModel
+import woowacourse.shopping.model.ProductUIModel
+
+class CartPresenter(
+    val view: CartContract.View,
+    private val repository: CartRepository,
+    offset: Int = 0,
+) : CartContract.Presenter {
+    private var offset = offset
+        set(value) {
+            field = when {
+                value < 0 -> 0
+                value > repository.getAll().size -> repository.getAll().size
+                else -> value
+            }
```

Comment:
[coreceIn](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.ranges/coerce-in.html) 함수를 활용해봐도 좋겠습니다~!

```suggestion
            field = value.coreceIn(0, repository.getAll().size)
```